test/functional/p2p_add_connections.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test add_outbound_p2p_connection test framework functionality"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.p2p import P2PInterface
+from test_framework.util import assert_equal
+
+
+class P2PAddConnections(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = False
+        self.num_nodes = 2
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_wallet()
+
+    def check_node_connections(self, num_node, num_in, num_out):
+        info = self.nodes[num_node].getnetworkinfo()
+        assert_equal(info["connections_in"], num_in)
+        assert_equal(info["connections_out"], num_out)
+
+    def run_test(self):
+        self.disconnect_nodes(0, 1)
+
+        self.log.info("Add 8 outbounds to node 0")
+        for i in range(8):
+            self.log.info("outbound: {}".format(i))
+            self.nodes[0].add_outbound_p2p_connection(P2PInterface(), connection_type="outbound")
```

Comment:
Is there any point in also testing calls like
```python
self.nodes[0].addconnection("127.0.0.1:36848", "outbound")
```

Reply:
`addconnection` currently doesn't have sanity tests to the RPC endpoint, but I'm leaving that as a future improvement since I'm trying to maximize reviewability and the added benefit seems small (esp since its a test-only RPC endpoint). this future improvement is tracked in https://github.com/bitcoin/bitcoin/pull/19315#issuecomment-735282520